bundles/org.openhab.binding.fmiweather/src/main/java/org/openhab/binding/fmiweather/internal/ForecastWeatherHandler.java

```diff
@@ -0,0 +1,238 @@
+        CHANNEL_TO_FORECAST_FIELD_NAME_AND_UNIT.put(channelId,
+                new AbstractMap.SimpleImmutableEntry<>(requestField, unit));
+    }
+
+    static {
+        addMapping(CHANNEL_TEMPERATURE, PARAM_TEMPERATURE, CELSIUS);
+        addMapping(CHANNEL_HUMIDITY, PARAM_HUMIDITY, PERCENT);
+        addMapping(CHANNEL_WIND_DIRECTION, PARAM_WIND_DIRECTION, DEGREE_ANGLE);
+        addMapping(CHANNEL_WIND_SPEED, PARAM_WIND_SPEED, METRE_PER_SECOND);
+        addMapping(CHANNEL_GUST, PARAM_WIND_GUST, METRE_PER_SECOND);
+        addMapping(CHANNEL_PRESSURE, PARAM_PRESSURE, MILLIBAR);
+        addMapping(CHANNEL_PRECIPITATION_INTENSITY, PARAM_PRECIPITATION_1H, MILLIMETRE_PER_HOUR);
+        addMapping(CHANNEL_TOTAL_CLOUD_COVER, PARAM_TOTAL_CLOUD_COVER, PERCENT);
+        addMapping(CHANNEL_FORECAST_WEATHER_ID, PARAM_WEATHER_SYMBOL, null);
+    }
+
+    private @NonNullByDefault({}) LatLon location;
+
+    public ForecastWeatherHandler(Thing thing) {
+        super(thing);
+        // Override poll interval to slower value
+        pollIntervalSeconds = (int) TimeUnit.MINUTES.toSeconds(QUERY_RESOLUTION_MINUTES);
+    }
+
+    @Override
+    public void initialize() {
+        try {
+            Object location = getConfig().get(BindingConstants.LOCATION);
+            if (location == null) {
+                throw new IllegalStateException();
+            }
+            String latlon = location.toString();
+            String[] split = latlon.split(",");
+            if (split.length != 2) {
+                throw new NumberFormatException(String.format(
+                        "Expecting location parameter to have latitude and longitude separated by comma (LATITUDE,LONGITUDE). Found %d values instead.",
+                        split.length));
+            }
+            this.location = new LatLon(new BigDecimal(split[0].trim()), new BigDecimal(split[1].trim()));
+            super.initialize();
+        } catch (IllegalStateException e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    String.format("location parameter not set %s %s", e.getClass().getName(), e.getMessage()));
+        } catch (NumberFormatException | IndexOutOfBoundsException e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    String.format("location parameter should be in format LATITUDE,LONGITUDE. Error: %s %s",
+                            e.getClass().getName(), e.getMessage()));
+        }
+    }
+
+    @Override
+    public void dispose() {
+        super.dispose();
+        this.location = null;
+    }
+
+    @Override
+    protected void update(int retry) {
+        if (location == null) {
+            return;
+        }
+        if (retry < RETRIES) {
+            try {
+                long now = Instant.now().getEpochSecond();
+                response = client.query(new ForecastRequest(location, floorToEvenMinutes(now, QUERY_RESOLUTION_MINUTES),
```

Comment:
updateChannels is also called with REFRESH command -- we use cached response in that case. In case response is updated in the meanwhile (if this is what you referred with unsafge), I don't see it a problem really.

Reply:
Given it is also called from another path it is a false-positive 😄